src/cat-data-service/src/state/mod.rs

```diff
@@ -15,12 +15,22 @@ pub struct State {
     /// This is Private, it needs to be accessed with a function.
     //event_db_handle: Arc<ArcSwap<Option<dyn EventDbQueries>>>, // Private need to get it with a function.
     pub event_db: Arc<dyn EventDbQueries>, // This needs to be obsoleted, we want the DB to be able to be down.
+    pub delay_seconds: u64,
     #[cfg(feature = "jorm-mock")]
     pub jorm: std::sync::Mutex<jorm_mock::JormState>,
 }
 
 impl State {
+    #[allow(dead_code)]
     pub async fn new(database_url: Option<String>) -> Result<Self, Error> {
```

Comment:
I think it would be better to leave only one function `new` but with an argument
`delay_seconds: Option<u64>`, so if it is `None` you can initialise with the default value ` RETRY_AFTER_DELAY_SECONDS_DEFAULT`

Reply:
@Mr-Leshiy I've done as you suggested.